src/04-test-class/index.test.ts

```diff
@@ -1,44 +1,84 @@
 // Uncomment the code below and write your tests
-// import { getBankAccount } from '.';
+import {
+  getBankAccount,
+  BankAccount,
+  InsufficientFundsError,
+  SynchronizationFailedError,
+} from '.';
 
 describe('BankAccount', () => {
+  let sut: BankAccount;
+  let intial: number;
+  let amount: number;
+  let appropriate_amount: number;
+
+  beforeEach(() => {
+    intial = 100;
+    sut = getBankAccount(intial);
+    amount = 150;
+    appropriate_amount: 50;
+  });
+
   test('should create account with initial balance', () => {
-    // Write your test here
+    expect(sut).toBeInstanceOf(BankAccount);
+    expect(sut.getBalance()).toBe(intial);
   });
 
   test('should throw InsufficientFundsError error when withdrawing more than balance', () => {
-    // Write your test here
+    expect(() => sut.withdraw(amount)).toThrowError(InsufficientFundsError);
   });
 
   test('should throw error when transferring more than balance', () => {
-    // Write your test here
+    const toAccount = getBankAccount(intial);
+    expect(() => sut.transfer(amount, toAccount)).toThrowError();
   });
 
   test('should throw error when transferring to the same account', () => {
-    // Write your test here
+    expect(() => sut.transfer(amount, sut)).toThrowError();
   });
 
   test('should deposit money', () => {
-    // Write your test here
+    expect(sut.deposit(amount).getBalance()).toBe(intial + amount);
   });
 
   test('should withdraw money', () => {
-    // Write your test here
+    expect(sut.withdraw(appropriate_amount).getBalance()).toBe(
+      intial - appropriate_amount,
+    );
   });
 
   test('should transfer money', () => {
-    // Write your test here
+    const toAccount = getBankAccount(intial);
+    expect(sut.transfer(appropriate_amount, toAccount).getBalance()).toBe(
+      intial - appropriate_amount,
+    );
+    expect(toAccount.getBalance()).toBe(intial + appropriate_amount);
   });
 
   test('fetchBalance should return number in case if request did not failed', async () => {
-    // Write your tests here
+    const result = await sut.fetchBalance();
+    if (result === null) {
+      expect(result).toBeNull();
+    } else {
+      expect(typeof result).toBe('number');
+    }
```

Comment:
```suggestion
    lodash.random = jest.fn(() => 1).mockImplementationOnce(() => 10);
    const bank = getBankAccount(20);
    const balance = await bank.fetchBalance();
    expect(typeof balance).toBe('number');
    jest.unmock('lodash');
```

Reply:
Thank you for carefully and thoroughly checking the work and making suggestions!
While I switched, I did not understand the whole idea, I think, together with the check will become clear.